### DISCUSSION TITLE
Turbo Repo in CI

### DISCUSSION THREAD

**Discussion by trm217** [vercel/turborepo#1457] [Help] [answered]
Hi everyone

I have recently been thinking about automating more tasks in a turborepo that contains three Next.js apps. 
Now I came to wonder, is there a way to get a list of all apps/packages inside a turborepo that have changed relative to another branch?

Imagine the following setup: We have a turborepo setup containing a package `chart-ui` and two apps `chart-editor` and `docs`, with `chart-editor` having `chart-ui` as a dependency.

Now let's say I update the source of package `chart-ui` and now want to run a bundle-analysis automation for all apps afflicted by changes.

So in this case we would only want `chart-editor` to run the bundle-analysis automation, since it's affected by the changes made to `chart-ui` but we wouldn't want to run the automation for the `docs` app.

How did you guys solve this in your mono-repos?

Edit:
I am aware that I can run scripts for changed apps/packges using --filter, however I can't seem to find a way to retrieve just a list of the apps

**Comment by awxalbert** [comment]
Need the exact same thing to customize CI rather than rely on turbo's built-in functionality.

**Reply by petermekhaeil** [reply, marked as answer]
`--dry-run` display details about the affected packages. Does this help?

https://turborepo.org/docs/reference/command-line-reference#--dry----dry-run
https://turborepo.org/blog/turbo-1-2-0#debug-and-automate-with---dry-run
